pwiz_tools/Skyline/Executables/SkylineBatch/SkylineBatch/MainForm.cs

```diff
@@ -0,0 +1,543 @@
+        }
+
+        private void btnExport_Click(object sender, EventArgs e)
+        {
+
+            var dialog = new SaveFileDialog { Title = Resources.Save_configurations, Filter = Resources.XML_file_extension };
+            if (dialog.ShowDialog(this) != DialogResult.OK) return;
+            configManager.ExportAll(dialog.FileName);
+        }
+
+
+        #endregion
+
+
+        #region Logging
+        // TODO: see if you can move some of this to skylineBatchLogger for better testing
+
+        private void comboLogList_SelectedIndexChanged(object sender, EventArgs e)
+        {
+            configManager.SelectLog(comboLogList.SelectedIndex);
+            SwitchLogger();
+        }
+
+        private async void SwitchLogger()
+        {
+            textBoxLog.Clear();
+
+            var logger = configManager.GetSelectedLogger();
+            try
+            {
+                await Task.Run(() =>
+                {
+                    // Read the log contents and display in the log tab.
+                    logger.DisplayLog();
+                });
+            }
+            catch (Exception ex)
+            {
+                ShowErrorDialog("Error Reading Log", ex.Message);
+            }
+
+            ScrollToLogEnd();
+        }
+
+        private void ScrollToLogEnd()
+        {
+            RunUi(() =>
+            {
+                textBoxLog.SelectionStart = textBoxLog.Text.Length;
+                textBoxLog.ScrollToCaret();
+            });
+        }
+
+        private void btnDeleteLogs_Click(object sender, EventArgs e)
+        {
+            var manageLogsForm = new LogForm(configManager);
+            manageLogsForm.StartPosition = FormStartPosition.CenterParent;
+            manageLogsForm.ShowDialog();
+        }
+
+        public void LogToUi(string text, bool scrollToEnd, bool trim)
+        {
+
+            RunUi(() =>
+            {
+                if (comboLogList.SelectedIndex != 0) return; // don't log if old log is displayed
+                if (trim)
+                {
+                    TrimDisplayedLog();
+                }
+
+                textBoxLog.AppendText(text);
+                textBoxLog.AppendText(Environment.NewLine);
+
+                if (!scrollToEnd) return;
+
+                ScrollToLogEnd();
+            });
+
+        }
+
+        private void TrimDisplayedLog()
+        {
+            var numLines = textBoxLog.Lines.Length;
+            const int buffer = SkylineBatchLogger.MaxLogLines / 10;
+            if (numLines > SkylineBatchLogger.MaxLogLines + buffer)
+            {
+                textBoxLog.ReadOnly = false; // Make text box editable. This is required for the following to work
+                textBoxLog.SelectionStart = 0;
+                textBoxLog.SelectionLength =
+                    textBoxLog.GetFirstCharIndexFromLine(numLines - SkylineBatchLogger.MaxLogLines);
+                textBoxLog.SelectedText = string.Empty;

```

Comment:
Usually you would set textBoxLog.Text to whatever you want the text to be, instead of using "SelectedText" to edit the text that is already in there.
(The danger I foresee with manipulating the text like this is that the TextBox might disagree with you about how many characters are there if there is anything unusual about the line ending characters).